CRM/Civicase/Service/CaseCategoryFromUrl.php

```diff
@@ -159,6 +165,24 @@ private function getCaseCategoryFromActivityIdInUrl($activityIdParamName) {
     }
   }
 
+  /**
+   * Return an Activity ID from an array received on the query string.
+   *
+   * @param string $activityIdParamName
+   *   Activity ID param name.
+   *
+   * @return int|null
+   *   The first activity ID found, or null.
+   */
+  private function getActivityIdFromArrayInUrl($activityIdParamName) {
+    $activityIds = CRM_Utils_Array::value($activityIdParamName, $_GET);
+    if ($activityIds) {
+      return array_shift($activityIds);
```

Comment:
if we need only one activity id then what is the reason for changing the frontend and backend code to work with an array

Reply:
Before these changes, there were places in which we were sending an array in `activity_ids`, and others in which we sent a single integer on `activity_id`. Also, the unit tests for frontend were using an array.
On the backend, as I commented on the PR, the backend method was expecting an array.
I decided to normalize using the array option. It is the less damaging option, considering that I can easily cast to array a single value for keeping the backend functionality (but the inverse is not true, "cast" an array to a single value).